module/signer.go

```diff
@@ -29,6 +32,33 @@ type ThresholdSigner interface {
 	Reconstruct(size uint, shares []crypto.Signature, indices []uint) (crypto.Signature, error)
 }
 
+var (
+	// DKGIncompleteError indicates that the node did not manage to obtain beacon keys from DKG at a certain view
+	DKGIncompleteError = errors.New("incomplete dkg")
+)
+
+// TODO: to be replaced by RandomBeaconSignerStore
+// ThresholdSignerStore returns the threshold signer object by view
 type ThresholdSignerStore interface {
+	// It returns:
+	//  - (signer, nil) if the node has beacon keys in the epoch of the view
+	//  - (nil, DKGIncompleteError) if the node doesn't have beacon keys in the epoch of the view
+	//  - (nil, error) if there is any exception
 	GetThresholdSigner(view uint64) (ThresholdSigner, error)
 }
+
+// MsgSigner signs a given message and produces a signature
+// TODO: could be renamed to Signer once the original Signer is replaced with MsgSigner
+type MsgSigner interface {
```

Comment:
The specific hasher we're using for BLS is concurrent safe [It's sad to see you leaving, but I'm sure ](https://github.com/onflow/flow-go/blob/master/crypto/hash/kmac.go#L147-L150). 
This doesn't apply to all hashers though. If we're gonna take advantage fo the hasher concurrency here, it would be good to mention it in a comment.